Database/Repositories/Implementations/TournamentsRepository.cs

```diff
@@ -185,6 +185,90 @@ public async Task DeletePooledBeatmapsAsync(int id, ICollection<int> beatmapIds)
     }
 
+    public async Task RerunAutomationChecks(int id, bool force = false)
+    {
+        Tournament? tournament = await TournamentsBaseQuery()
+            .FirstOrDefaultAsync(t => t.Id == id);
+
+        if (tournament is null)
+        {
+            return;
+        }
+
+        var update = force || (tournament.VerificationStatus != VerificationStatus.Rejected &&
+                               tournament.VerificationStatus != VerificationStatus.Verified);
+
+        if (update)
+        {
+            tournament.VerificationStatus = VerificationStatus.None;
+            tournament.RejectionReason = TournamentRejectionReason.None;
+            tournament.ProcessingStatus = TournamentProcessingStatus.NeedsAutomationChecks;
+        }
+
+        ResetMatchChecks(tournament, force);
+        await UpdateAsync(tournament);
+    }
+
+    private static void ResetMatchChecks(Tournament tournament, bool force)
+    {
+        foreach (Match match in tournament.Matches)
+        {
+            var matchUpdate = force || (match.VerificationStatus != VerificationStatus.Rejected &&
+                                        match.VerificationStatus != VerificationStatus.Verified);
+
+            if (!matchUpdate)
+            {
+                continue;
+            }
+
+            match.VerificationStatus = VerificationStatus.None;
+            match.WarningFlags = MatchWarningFlags.None;
+            match.RejectionReason = MatchRejectionReason.None;
+            match.ProcessingStatus = MatchProcessingStatus.Done;
+
+            ResetGameChecks(match, force);
+        }
+    }
+
+    private static void ResetGameChecks(Match match, bool force)
+    {
+        foreach (Game game in match.Games)
+        {
+            var gameUpdate = force || (game.VerificationStatus != VerificationStatus.Rejected &&
+                                        game.VerificationStatus != VerificationStatus.Verified);
+
+            if (!gameUpdate)
+            {
+                continue;
+            }
+
+            game.VerificationStatus = VerificationStatus.None;
+            game.WarningFlags = GameWarningFlags.None;
+            game.RejectionReason = GameRejectionReason.None;
+            game.ProcessingStatus = GameProcessingStatus.Done;
+
+            ResetScoreChecks(game, force);
+        }
+    }
+
+    private static void ResetScoreChecks(Game game, bool force)
+    {
+        foreach (GameScore score in game.Scores)
+        {
+            var scoreUpdate = force || (score.VerificationStatus != VerificationStatus.Rejected &&
+                                        score.VerificationStatus != VerificationStatus.Verified);
+
+            if (!scoreUpdate)
+            {
+                continue;
+            }
+
+            score.VerificationStatus = VerificationStatus.None;
+            score.RejectionReason = ScoreRejectionReason.None;
+            score.ProcessingStatus = ScoreProcessingStatus.Done;
+        }
+    }
+
```

Comment:
These should be either extension methods or defined on their appropriate repositories, not all on the tournaments repo